### PR TITLE
Deploy electricityMap static assets to GCS

### PR DESCRIPTION
**Pull request by corradio**
Internal PR to simplify deployment on our side:
- upload assets to GCS
- use a consistent working directory such that Dockerfile and BUILD.yaml produce the same folder structure

~Blocked by https://github.com/tmrowco/brick/pull/74~

### CODE REVIEW

File: web/BUILD.yaml [hunk 1]
```diff
@@ -9,18 +9,36 @@ steps:
...
+      - ./{.babelrc,.eslintrc,server.js,webpack.config.js}
     commands:
       - mkdir -p public/dist
-      - mkdir -p src
       - bash topogen.sh
+      - yarn build-release
     outputs:
-      - public/dist/zonegeometries.json
+      - src/world.json
+      - public/dist
```
Comment (corradio) [inline web/BUILD.yaml:34]: brick now also builds the frontend

File: web/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FROM node:12.13.1
-WORKDIR /home/web
+WORKDIR /home/src/electricitymap/contrib/web
```
Comment (corradio) [inline web/Dockerfile:2]: in order to keep consistent with brick paths

File: web/server.js [hunk 3]
```diff
@@ -207,14 +197,22 @@ app.use('/', (req, res) => {
       // Make the paths absolute as that's required for BrowserHistory routing
       // to work normally and it's also ok when used with the https:// protocol
       // as resources are mounted to a fixed location.
-      resolvePath: function(relativePath) { return '/' + relativePath; },
+      // Note: `resolvePath` is executed on the client as well,
+      // as it is used in react components. We can't therefore include any variables
+      // in its closure. It would be better to pass a `pathPrefix` instead.
+      resolvePath: (!isProduction || isStaging)
```
Comment (corradio) [inline web/server.js:203]: this is the important non-lint change to make sure we use GCS in production

File: web/server.js [hunk 4]
```diff
@@ -224,7 +222,13 @@ app.use('/', (req, res) => {
   }
 });
 
+if (isProduction) {
+  app.get('/*', (req, res) =>
```
Comment (corradio) [inline web/server.js:226]: legacy redirect (but shouldn't be necessary)

**Review by FelixDQ — APPROVED** [review]
Nice 💪

**Pull request merged**